src/Microsoft.DotNet.Interactive/PackageRestoreContext.cs

```diff
@@ -54,7 +57,13 @@ private IEnumerable<string> NativeProbingRoots()
             }
         }
 
-        public void AddRestoreSource(string source) => _restoreSources.Add(source);
+        public void AddRestoreSource(string source)
+        {
+            if (!_resolvedRestoreSources.TryGetValue(source, out var _))
```

Comment:
You shouldn't need the `if` condition given you're using `TryAdd`.

Reply:
The if allows me to add to requested, if it's not already in resolved.

Reply:
Oh, I misread these as the same dictionary. In this case, we've introduced a potential concurrency issue here, though it's very unlikely given that the scheduler synchronizes command invocation. I wonder if consolidating these into a single collection would simplify things.

Reply:
changed to use GetOrAdd()